app/api/routes/user.py

```diff
@@ -10,29 +19,241 @@
+
+    # Update other fields
+    for field, value in update_data.items():
+        if value is not None:
+            user_data[field] = value
+
+    # Update the 'updated_at' timestamp
+    user_data["updated_at"] = datetime.now(timezone.utc)
+
+    # Update the user in the database
+    users_db[str(user_id)] = user_data
+
+    # Step 5: Return Updated User Data
+    updated_user = User(**user_data)
+    return UpdatedUserResponseModel(
+        id=updated_user.id,
+        username=updated_user.username,
+        email=updated_user.email,
+        is_admin=updated_user.is_admin,
+        is_active=updated_user.is_active,
+        created_at=updated_user.created_at,
+        updated_at=updated_user.updated_at,
+    )
+
+
+@router.get("/{user_id}", response_model=CreateUserResponseModel)
+async def get_user_details(
+    user_id: UUID, current_user: User = Depends(get_current_user)
+):
+    # Step 1: Authenticate User
+    if not current_user.get("is_admin") and str(current_user.get("id")) != str(user_id):
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="Access denied. You can only view your own details.",
+        )
+
+    # Step 2: Validate User ID
+    user_data = users_db.get(str(user_id))
+    if not user_data:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="User not found"
+        )
+
+    # Step 3: Retrieve User Data
+    user = User(**user_data)
+
+    # Step 4: Return User Data
+    return CreateUserResponseModel(
+        id=user.id,
+        username=user.username,
+        email=user.email,
+        is_admin=user.is_admin,
+        is_active=user.is_active,
+        created_at=user.created_at,
+        updated_at=user.updated_at,
     )
 
+
+@router.get("/", response_model=list[dict], status_code=status.HTTP_200_OK)
+async def get_all_users(current_admin: User = Depends(get_current_admin_user)):
+
+    try:
+        all_users = list(users_db.values())
+
+        return [
+            {
+                "id": user["id"],
+                "username": user["username"],
+                "email": user["email"],
+                "is_admin": user["is_admin"],
+                "is_active": user["is_active"],
+                "created_at": user["created_at"],
+                "updated_at": user["updated_at"],
+            }
+            for user in all_users
+        ]
+
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"An error occurred while retrieving users: {str(e)}",
+        )
+
+
+@router.get("/{user_id}/orders", status_code=status.HTTP_200_OK)
+async def get_orders_for_user(
+    user_id: UUID, current_user: User = Depends(get_current_user)
+):
+    # Authenticate User
+    if not current_user:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED, detail="Unauthorized"
+        )
+
+    # Validate User ID
```

Comment:
Authentication and authorization should already be implemented in the get_current_user function!!!

Reply:
done